pkg/credentials/repository.go

```diff
@@ -2,36 +2,60 @@ package credentials
 
 import (
 	"context"
-	"math"
+	"time"
 
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/encoding/gzip"
+	"google.golang.org/grpc/status"
 
-	agentclient "github.com/kubeshop/testkube/pkg/agent/client"
-	"github.com/kubeshop/testkube/pkg/cloud"
+	"github.com/kubeshop/testkube/pkg/controlplaneclient"
+	"github.com/kubeshop/testkube/pkg/log"
 )
 
+type grpcstatus interface {
+	GRPCStatus() *status.Status
+}
+
+const (
+	GetCredentialRetryCount = 30
+)
+
+func getIterationDelay(iteration int) time.Duration {
+	if iteration < 5 {
+		return 500 * time.Millisecond
+	} else if iteration < 100 {
+		return 1 * time.Second
+	}
+	return 5 * time.Second
```

Comment:
should it be const?

Reply:
In this case I treat this function as a const - alternatively I could create struct like:
```go
struct RepetitionRule {
  Delay time.Duration
  MaxIteration uint32
}

var repetitions = []RepetitionRule{
  {Delay: 500 * time.Millisecond, MaxIteration: 5},
  {Delay: 1 * time.Second, MaxIteration: 100},
  {Delay: 5 * time.Second, MaxIteration: math.MaxUint32},
}
```

And handling that. I think that the current solution is simpler, but I'm open to refactor (or maybe some better way?)

Reply:
ok, let's use current solution